### PR TITLE
feat: [PL-37913]: Added delegate meta data in delegate selection logs

### PR DESCRIPTION
**Pull request by meenakshi-harness**
## Describe your changes
We are starting to persist delegate name and id as part for delegate selection logs. This will be used by pipeline to fetch delegate details by taskID.

Testing:
1. Create a task for connector validation.
2. verified data in delegateLogSelection collection  

Screenshot 2023-07-13 at 11 23 52 AM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50403)

### CODE REVIEW

**Comment by jennyjames21** [comment]
LGTM, Please add summary and test section

File: 400-rest/src/main/java/software/wings/service/impl/DelegateSelectionLogsServiceImpl.java [hunk 1]
```diff
@@ -220,13 +221,16 @@ public void logTaskAssigned(String delegateId, DelegateTask delegateTask) {
       return;
     }
     Delegate delegate = delegateCache.get(delegateTask.getAccountId(), delegateId, false);
+    DelegateMetaInfo delegateMetaInfo = null;
```
Comment (xingchi-jin) [inline 400-rest/src/main/java/software/wings/service/impl/DelegateSelectionLogsServiceImpl.java:224]: Overall looks good. One point i want to make is
Can we not use the "DelegateMetaInfo" pojo in the DelegateSelectionLog persistence. The reason is, the same pojo has been used in various delegate response DTOs. It just causes backward compatibility issue unnecessarily between task response and DelegateSelectionLog persistence. 
Here what needed is just hostname and delegate id . We can create a new class or just put the two fields in it

**Comment by xingchi-jin** [comment]
pl. update description and test results. Thanks for making the change !

File: 400-rest/src/main/java/software/wings/service/impl/DelegateSelectionLogsServiceImpl.java (hunk 1 above)
Comment (meenakshi-harness) [reply 400-rest/src/main/java/software/wings/service/impl/DelegateSelectionLogsServiceImpl.java:224]: Sure, let me create a new class.

Comment (meenakshi-harness) [reply 400-rest/src/main/java/software/wings/service/impl/DelegateSelectionLogsServiceImpl.java]: @xingchi-jin @jennyjames21 on testing, looks like we needed few changes while building DelegateSelectionLogParams as well. Can you re-review.

**Comment by meenakshi-harness** [comment]
trigger uts

**Review by jennyjames21 — DISMISSED** [review]

**Comment by meenakshi-harness** [comment]
trigger messagecheck

**Comment by meenakshi-harness** [comment]
trigger uts

**Review by jennyjames21 — APPROVED** [review]

File: 420-delegate-service/src/main/java/io/harness/selection/log/DelegateMetaData.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.Data;
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.FieldNameConstants;
+
+@Data
+@Builder
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@FieldNameConstants(innerTypeName = "DelegateMetaDataKeys")
+@OwnedBy(DEL)
+public class DelegateMetaData {
```
Comment (xingchi-jin) [inline 420-delegate-service/src/main/java/io/harness/selection/log/DelegateMetaData.java:25]: add a comment to this class, where it's used

**Review by xingchi-jin — APPROVED** [review]
A nit comment, and approve

File: 420-delegate-service/src/main/java/io/harness/selection/log/DelegateMetaData.java (hunk 2 above)
Comment (meenakshi-harness) [reply 420-delegate-service/src/main/java/io/harness/selection/log/DelegateMetaData.java:25]: Sure will add in seperate PR

**Pull request merged**